src/nupic/algorithms/Cells4.cpp

```diff
@@ -67,18 +67,39 @@ static nupic::Timer chooseCellsTimer;
 
 #endif
 
-Cells4::Cells4(UInt nColumns, UInt nCellsPerCol, UInt activationThreshold,
-               UInt minThreshold, UInt newSynapseCount,
-               UInt segUpdateValidDuration, Real permInitial,
-               Real permConnected, Real permMax, Real permDec, Real permInc,
-               Real globalDecay, bool doPooling, int seed, bool initFromCpp,
-               bool checkSynapseConsistency)
-    : _rng(seed) {
+Cells4::Cells4(UInt nColumns, UInt nCellsPerCol,
+                     UInt activationThreshold,
+                     UInt minThreshold,
+                     UInt newSynapseCount,
+                     UInt segUpdateValidDuration,
+                     Real permInitial,
+                     Real permConnected,
+                     Real permMax,
+                     Real permDec,
+                     Real permInc,
+                     Real globalDecay,
+                     bool doPooling,
+                     int seed,
+                     bool initFromCpp,
+                     bool checkSynapseConsistency)
+  : _rng(seed < 0 ? rand() : seed)
```

Comment:
unnecessary change for this PR? May it break the other 3 tests? 
// I gues it complies with the API?

Reply:
It looks so much more readable.